table/block_based/block_based_table_reader.cc

```diff
@@ -2384,7 +2384,13 @@ void BlockBasedTable::RetrieveMultipleBlocks(
       }
     }
     if (s.ok()) {
-      if (options.fill_cache) {
+      Cache* block_cache = rep_->table_options.block_cache.get();
```

Comment:
I feel it might be cleaner to call `MaybeReadBlockAndLoadToCache` unconditionally here and check whether `block_entry->GetValue()` is still `nullptr` after the call (instead of duplicating the checks in `MaybeReadBlockAndLoadToCache` in order to figure whether the call would be a NOOP or not).